pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,198 @@
+import pymc as pm
+import numpy as np
+import pytensor.tensor as at
+import pytensor
+
+from pymc.distributions.distribution import (
+    Distribution,
+    Discrete,
+    SymbolicRandomVariable,
+    _moment,
+)
+
+from pymc.logprob.abstract import _logprob
+from pymc.pytensorf import intX
+from pytensor.tensor import TensorVariable
+from pytensor.tensor.random.op import RandomVariable
+from pytensor.graph.basic import Node
+
+from pymc.distributions.shape_utils import (
+    _change_dist_size,
+    get_support_shape_1d,
+)
+
+
+def validate_transition_matrix(P):
```

Comment:
Instead of eagerly evaluating P,  you can wrap it in a PyTensor assert

Reply:
I followed the example set other distributions and wrapped the `logp` return with `check_parameters`, which no longer requires eager evaluation.